qiskit/providers/ibmq/experiment/experimentservice.py

```diff
@@ -274,20 +287,32 @@ def analysis_results(
 
         Returns:
             A list of analysis results.
+
+        Raises:
+            ValueError: If an invalid parameter value is specified.
         """
+        if limit is not None and (not isinstance(limit, int) or limit <= 0):  # type: ignore
+            raise ValueError(f"{limit} is not a valid `limit`, which has to be a positive integer.")
+
         qualit_list = []
         if quality:
             for op, qual in quality:
                 if isinstance(qual, ResultQuality):
                     qual = qual.value
                 qual_str = qual if op == 'eq' else "{}:{}".format(op, qual)
                 qualit_list.append(qual_str)
-        response = self._api_client.analysis_results(
-            backend_name=backend_name, device_components=device_components,
-            experiment_uuid=experiment_id, result_type=result_type, quality=qualit_list)
         results = []
-        for result in response:
-            results.append(AnalysisResult.from_remote_data(result))
+        marker = None
+        while limit and limit > 0:
```

Comment:
Same as above, what if limit is None? Do you get anything back? Do you get *everything* back?

Reply:
Yeah this PR was not as done as I thought. It should be fixed now.